File: templates/dotnet-teams/content/teams-msgext-search/Teams.MsgExt.Search/Search/SearchApp.cs
using AdaptiveCards;
using AdaptiveCards.Templating;
using Microsoft.Bot.Builder;
using Microsoft.Bot.Builder.Teams;
using Microsoft.Bot.Schema;
using Microsoft.Bot.Schema.Teams;
using Newtonsoft.Json;

namespace Teams.MsgExt.Search.Search;

public class SearchApp : TeamsActivityHandler
{
    protected override async Task<MessagingExtensionResponse> OnTeamsMessagingExtensionQueryAsync(ITurnContext<IInvokeActivity> turnContext, MessagingExtensionQuery query, CancellationToken cancellationToken)
    {
        var text = query?.Parameters?[0]?.Value as string ?? string.Empty;

        var card = await File.ReadAllTextAsync(Path.Combine(".", "Resources", "card.json"), cancellationToken);
        var template = new AdaptiveCardTemplate(card);

        return new MessagingExtensionResponse
        {
            ComposeExtension = new MessagingExtensionResult
            {
                Type = "result",
                AttachmentLayout = "list",
                Attachments = [
                    new MessagingExtensionAttachment
                        {
                            ContentType = AdaptiveCard.ContentType,
                            Content = JsonConvert.DeserializeObject(template.Expand(new { title = text })),
                            Preview = new ThumbnailCard { Title = text }.ToAttachment()
                        }
                ]
            }
        };
    }
}

File: templates/nodejs-teams-msgext-search/templates/ts-ttk/src/searchApp/searchApp.ts
import {
  TeamsActivityHandler,
  CardFactory,
  TurnContext,
  MessagingExtensionQuery,
  MessagingExtensionResponse,
  AttachmentLayoutTypes,
} from "botbuilder";
import * as ACData from "adaptivecards-templating";
import card from "../cards/card.json";

export class SearchApp extends TeamsActivityHandler {

  public async handleTeamsMessagingExtensionQuery(
    context: TurnContext,
    query: MessagingExtensionQuery
  ): Promise<MessagingExtensionResponse> {

    const text = query.parameters[0]?.value as string ?? "";
    var template = new ACData.Template(card);

    return {
      composeExtension: {
        type: 'result',
        attachmentLayout: AttachmentLayoutTypes.List,
        attachments: [
          {
            contentType: 'application/vnd.microsoft.card.adaptive',
            content: template.expand({ $root: { text } }),
            preview: CardFactory.thumbnailCard(text)
          },
        ],
      },
    };
  }
}
